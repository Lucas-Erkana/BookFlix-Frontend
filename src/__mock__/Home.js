import React from 'react';
import { BrowserRouter } from 'react-router-dom';

const Home = () => {
  const cards = [{
    name: 'Movie 1',
    image: 'https://res.cloudinary.com/dx0nauane/image/upload/v1624349663/Movie%20Card%20Images/Movie_1_1_xqjz3c.jpg',
    details: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla euismod, sapien vitae aliquam dapibus, magna quam ultricies nunc, nec tincidunt nisl nunc vitae elit. ',
    price: 100,
    id: 1,
  }];

  const isMobile = window.innerWidth <= 768;

  return (
    <BrowserRouter>
      <div className="home col-md col">
        <h1 className="bold-font homepage-heading">
          OUR SERVICES
        </h1>
        <p className="gray-font">
          Please select a Movie to reserve
        </p>
        <hr className="dash home-dash" />
        <div className="carousel-container col-12">

          <div>
            {cards.map((card) => (
              <div
                key={card.id}
              >
                <div>
                  {card.name}
                  {card.image}
                  {card.details}
                  {card.price}
                  {card.id}
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    </BrowserRouter>
  );
};

export default Home;
