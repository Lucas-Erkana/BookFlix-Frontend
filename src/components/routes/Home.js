import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Navigation, A11y } from 'swiper';
import { Swiper, SwiperSlide } from 'swiper/react';
import 'swiper/css';
import 'swiper/css/navigation';
import 'swiper/css/pagination';
import 'swiper/css/scrollbar';
import MovieCard from '../child-components/MovieCard';
import { fetchMovies } from '../../redux/MoviesSlice';

const Home = () => {
  const dispatch = useDispatch();
  const cards = useSelector((state) => state.movies.movies);

  useEffect(() => {
    dispatch(fetchMovies());
  }, [dispatch]);

  const [isMobile, setIsMobile] = useState(window.innerWidth <= 968);
  const handleResize = () => {
    setIsMobile(window.innerWidth <= 768);
  };

  useEffect(() => {
    window.addEventListener('resize', handleResize);
    return () => {
      window.removeEventListener('resize', handleResize);
    };
  }, []);

  const slidesPerView = isMobile ? 1 : 3;

  return (
    <div className="home col-md col">
      <h1 className="bold-font homepage-heading text-uppercase">
        movies to watch
      </h1>
      <p className="gray-font">
        Please select a Movie to reserve
      </p>
      <hr className="dash home-dash" />
      <div className="carousel-container col-12">

        <Swiper
          className="movie-list col-10"
          modules={[Navigation, A11y]}
          spaceBetween={0}
          slidesPerView={slidesPerView}
          navigation
          loop // Enable looping
        >
          {/* Add slides at the beginning for a seamless loop */}
          {cards.slice(-slidesPerView).map((card) => (
            <SwiperSlide key={card.id}>
              <MovieCard
                name={card.name}
                image={card.image}
                details={card.details}
                price={card.price}
                id={card.id}
              />
            </SwiperSlide>
          ))}

          {/* Original slides */}
          {cards.map((card) => (
            <SwiperSlide key={card.id}>
              <MovieCard
                name={card.name}
                image={card.image}
                details={card.details}
                price={card.price}
                id={card.id}
              />
            </SwiperSlide>
          ))}

          {/* Add slides at the end for a seamless loop */}
          {cards.slice(0, slidesPerView).map((card) => (
            <SwiperSlide key={card.id}>
              <MovieCard
                name={card.name}
                image={card.image}
                details={card.details}
                price={card.price}
                id={card.id}
              />
            </SwiperSlide>
          ))}
        </Swiper>
      </div>

    </div>
  );
};

export default Home;
