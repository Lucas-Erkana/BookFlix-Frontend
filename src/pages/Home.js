import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Navigation, A11y } from 'swiper';
import { Swiper, SwiperSlide } from 'swiper/react';
import 'swiper/css';
import 'swiper/css/navigation';
import 'swiper/css/pagination';
import 'swiper/css/scrollbar';
import MovieCard from '../components/MovieCard';
import { fetchMovies } from '../store/MoviesSlice';

const Home = () => {
  const dispatch = useDispatch();
  const cards = useSelector((state) => state.movies.movies);

  useEffect(() => {
    dispatch(fetchMovies());
  }, [dispatch]);

  const isMobile = window.innerWidth <= 768;

  return (
    <div className="home col-md col">
      <h1 className="bold-font homepage-heading">
        OUR MOVIES
      </h1>
      <p className="gray-font">
        Please select a Movie to reserve
      </p>
      <hr className="dash home-dash" />
      <div className="carousel-container col-12">

        <Swiper
          className="movie-list col-10"
          modules={[Navigation, A11y]}
          spaceBetween={0}
          slidesPerView={isMobile ? 1 : 3}
          navigation
        >
          {cards.map((card) => (
            <SwiperSlide
              key={card.id}
            >
              <MovieCard
                name={card.name}
                image={card.image}
                details={card.details}
                price={card.price}
                id={card.id}
              />
            </SwiperSlide>
          ))}
        </Swiper>
      </div>

    </div>
  );
};

export default Home;
