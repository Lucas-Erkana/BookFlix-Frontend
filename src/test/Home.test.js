import { render, cleanup } from '@testing-library/react';
import Home from '../__mock__/Home';

afterEach(() => {
  cleanup();
});

describe('Home', () => {
  it('Home renders correctly', () => {
    const home = render(<Home />);
    expect(home).toMatchSnapshot();
  });
});
