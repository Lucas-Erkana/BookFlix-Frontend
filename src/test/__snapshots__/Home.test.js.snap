// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home col-md col"
      >
        <h1
          class="bold-font homepage-heading"
        >
          MOVIES TO WATCH
        </h1>
        <p
          class="gray-font"
        >
          Please select a Movie to reserve
        </p>
        <hr
          class="dash home-dash"
        />
        <div
          class="carousel-container col-12"
        >
          <div>
            <div>
              <div>
                Service 1
                https://res.cloudinary.com/dx0nauane/image/upload/v1624349663/Service%20Card%20Images/Service_1_1_xqjz3c.jpg
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla euismod, sapien vitae aliquam dapibus, magna quam ultricies nunc, nec tincidunt nisl nunc vitae elit. 
                100
                1
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home col-md col"
    >
      <h1
        class="bold-font homepage-heading"
      >
        MOVIES TO WATCH
      </h1>
      <p
        class="gray-font"
      >
        Please select a Movie to reserve
      </p>
      <hr
        class="dash home-dash"
      />
      <div
        class="carousel-container col-12"
      >
        <div>
          <div>
            <div>
              Service 1
              https://res.cloudinary.com/dx0nauane/image/upload/v1624349663/Service%20Card%20Images/Service_1_1_xqjz3c.jpg
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla euismod, sapien vitae aliquam dapibus, magna quam ultricies nunc, nec tincidunt nisl nunc vitae elit. 
              100
              1
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
